Add libcloud support for image guest os features.

illfelder opened the pull request:
Image "guestOsFeature" support during image copy and creation.
### Description

We are introducing a new optional property on the image resource known as "guestOsFeature". The feature is available in the alpha API and is needed to optimize virtual machine settings during instance creation. The only value currently supported (if specified) is "VIRTIO_SCSI_MULTIQUEUE".
### Status

Done, ready for review
### Checklist (tick everything that applies)
- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [x] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

illfelder committed:
Add libcloud support for image guest OS features.

Guest OS features are a repeated field in the API.

tonybaloney commented:
cc @supertom @erjohnso

supertom commented on libcloud/compute/drivers/gce.py:
```diff
@@ -4752,29 +4753,31 @@ def ex_copy_image(self, name, url, description=None, family=None,
 
-        :param  guest_os_feature: The features of the guest opertaing system.
-        :type   guest_os_feature: ``str``
+        :param  guest_os_features: The features of the guest opertaing system.
```
"operating"

illfelder replied:
Thanks!

supertom commented on libcloud/compute/drivers/gce.py:
```diff
@@ -2214,11 +2214,11 @@ def ex_create_image(self, name, volume, description=None, family=None,
-                                    values include \'VIRTIO_SCSI_MULTIQUEUE\'
-                                    if specified.
-        :type     guest_os_feature: ``str``
+        :keywork  guest_os_features: Features of the guest opertaing system,
```
"operating"

supertom commented:
LGTM, couple of spelling errors I noted.

tonybaloney commented on libcloud/compute/drivers/gce.py:
```diff
@@ -2243,7 +2251,14 @@ def ex_create_image(self, name, volume, description=None, family=None,
+        if guest_os_features:
+            image_data['guestOsFeatures'] = []
+            for feature in guest_os_features:
+                if feature in possble_features:
```
that should be `for feature in possible_features:` and also showing this case needs a test :-)

illfelder replied:
Yup, thanks for the note. I missed the typo (and failing test) when I refactored some of the code).

tonybaloney commented:
@illfelder 1 typo to correct and please also improve the test scenario to validate the outgoing payload.

illfelder committed:
Added testing for guest os features.

illfelder commented:
Added testing to validate the request is getting created properly. I also fixed a comment about the type of guest_os_features when creating an image.

tonybaloney commented:
@illfelder think you broke the build with the extra commit :-)

illfelder commented:
Yup, fixed the linter issue. Should be good now whenever travis next runs.

tonybaloney commented:
OK 👍
